### DISCUSSION TITLE
+list-actions prints garbage characters

### DISCUSSION THREAD

**Discussion by kaddkaka** [ghostty-org/ghostty#3755] [Feature Requests, Ideas] [answered]
First line of output looks strange with broken character:

```console
$ ghostty +list-actions
�~��Zignore
unbind
csi
esc
[...]
```

**Comment by kaddkaka** [comment]
It does not reproduce every time but I have gotten it more than once.

**Comment by 00-kat** [comment]
I remember seeing another discussion where someone consistently got this when trying to use the CLI, but I can't find it right now.

Edit: tried looking through the last multiple days, but I still can't find it :<.

**Comment by jaboatman** [comment]
I see this on any command actually, it seems to just show up semi randomly.
I have a bash alias `ff=firefox` and I got:
```bash
$ ff -v
��1[Mozilla Firefox 135.0
```

**Reply by 00-kat** [reply]
Does adding `shell-integration = none` to your Ghostty configuration change anything?

**Comment by wong-justin** [comment]
I reliably reproduced this while messing with multiline bash aliases:

```bash
$ alias garbage='echo start
> echo end'

$ garbage
start
�\��dend
```

```bash
$ alias normal='(echo start
> echo end)'

$ normal
start
end
```

Adding `shell-integration = none` seemed to make the garbage alias behave normal.

**Reply by 00-kat** [reply]
Yeah I can reproduce it with that too.

**Reply by 00-kat** [reply]
Commenting out this line: https://github.com/ghostty-org/ghostty/blob/38908e01268298ced33b7b6835e49b8c6d437e12/src/shell-integration/bash/ghostty.bash#L169 mostly fixes it for me.  I say mostly because `1` appears from somewhere too.
![A screenshot.](https://github.com/user-attachments/assets/aa689b0a-56d2-4964-a774-52b5281d2268)
(My `a` alias is your `garbage` alias.)

If I comment out both this: https://github.com/ghostty-org/ghostty/blob/38908e01268298ced33b7b6835e49b8c6d437e12/src/shell-integration/bash/ghostty.bash#L165 and the line I mentioned above, the garbage comes back.  Commenting out the previously mentioned line as well as any other `printf` in the file doesn't result in any issues, however (other than the `1` appearing).

(I'm not very familiar with the Bash shell integration so I'm not too sure what the issue is; I'm only commenting this in case someone familiar with it can glean anything from this observation.)

**Reply by kaddkaka** [reply]
I got the garbage with almost the same alias and this prompt defined by as:
```console
$ echo $PS1
${debian_chroot:+($debian_chroot)}\u@\h:\w\$
```

so multi-line prompt does not seem to be a prerequisite

**Reply by jparise** [reply]
I'm also able to reproduce this in its various flavors. It appears to be related to the way we're saving and restoring `PS0`. I'll open an issue (#7802) to track this further.

**Reply by jparise** [reply]
I think I have a solution for this in #7808.

**Comment by kaddkaka** [comment]
I'm still seeing this issue. With the repro above, should we upgrade this discussion to a bug?

**Reply by 00-kat** [reply]
cc @ghostty-org/shell.

**Reply by sokai** [reply]
FYI: https://github.com/atuinsh/atuin/issues/2789 + https://github.com/ghostty-org/ghostty/discussions/7639 #ty!

**Reply by kaddkaka** [reply]
How to start a bug issue?

**Reply by 00-kat** [reply]
The relevant maintainer for this subsystem has to see it and promote it to an issue.

**Reply by jparise** [reply]
Tracking this as #7802. Sorry for not promoting this discussion earlier.

**Comment by sokai** [comment]
Hey!

@kaddkaka (and/or anyone who is _affected_): Are you using any _additional prompt program_?

My _highly personalised_  [Starship](https://starship.rs/) config seems to generate _these characters_ (somehow) – ~~**but only**~~ w/ + w/o a **multiline** prompt ~~**(?)**~~ …

**Reply by kaddkaka** [reply]
I'm not using starship and I'm not using multi-line prompt. I've encountered the issue with an empty `PS1.` and without other things might mess with the prompt like direnv/zoxide.

my prompt_command includes this (which is from the shell_integration from ghostty I believe):
```console
$ echo $PROMPT_COMMAND
__bp_precmd_invoke_cmd
```

If I `unset PROMPT_COMMAND`, the problem goes away.

**Reply by sokai** [reply]
Thank you @kaddkaka !

I _checked_ again and _(have to) agree_ to all the _findings_ … :/

BTW: _fun_ while testing the `$PROMPT_COMMAND`
```sh
$ echo $PROMPT_COMMAND
bash: bad substitution: no closing "`" in `�
`�__bp_precmd_invoke_cmd starship_precmd :

$ echo $PROMPT_COMMAND
__bp_precmd_invoke_cmd starship_precmd :
```
But I couldn't _replicate_ the 1st output in the same terminal afterwards. So I have no _clue_ what _triggered_ that. The _only evidence_ seems time, because the terminal was inactive for several hours … _#anyway_

**Reply by jparise** [reply]
I'm not sure if this has a different root cause than the thread above, but I'm hopeful that #7808 will also fix these cases given that the unprintable output appears to come from the same place in the prompt sequence.

**Comment by sokai** [comment, marked as answer]
FYI: this should be fixed by https://github.com/ghostty-org/ghostty/pull/7808, which is released w/ [ghostty 1.2.0](https://github.com/ghostty-org/ghostty/milestone/5?closed=1) 🚀
(Not tested it out by myself, ATM, but refreshed my knowledge to this issue … ;))

**Reply by kaddkaka** [reply]
Thanks for making a note here, wouldn't have seen it otherwise. ❤️
